### PR TITLE
doc: Add layout config examples and split reference

### PR DESCRIPTION
**Pull request by mschwan-phytec**
Add layout configuration examples. One example contains a minimal file with just enough settings to write a simple layout to an SD card. The other one showcases all supported features.

Split the reference into its own file, so it is separated from the examples.

### CODE REVIEW

File: doc/layout-config-examples.rst [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+       type: primary
+       filesystem: ext4
+       size: 192MiB
+       input:
+         - uri: file://config-partition.tar.gz
+
+     - label: ROOT0
+       type: logical
+       filesystem: null
+       expand: true
```
Comment (landerweit-phytec) [inline doc/layout-config-examples.rst:101]: Could you add `block-size: 4KiB` here.

File: doc/layout-config-examples.rst [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+       input:
+         - uri: file://rootfs.ext4
+
+     - label: ROOT1
+       type: logical
+       filesystem: null
+       expand: true
```
Comment (landerweit-phytec) [inline doc/layout-config-examples.rst:108]: Same here

**Review by landerweit-phytec — CHANGES_REQUESTED** [review]
Looks overall good. Just one thing below.

File: doc/layout-config-examples.rst (hunk 1 above)
Comment (mschwan-phytec) [reply doc/layout-config-examples.rst:101]: Ah, yes. Thanks for noticing.

**Comment by mschwan-phytec** [comment]
I've moved the layout configuration examples to separate YAML files. I think this looks cleaner and more organized.

**Review by landerweit-phytec — APPROVED** [review]

**Pull request merged**